<?xml version="1.0"?>
<launch>
    <!--argument-->
    <arg name="low_lines" default="4" />
    <arg name="z_threshold" default="0.03" />
<!--    <arg name="z_low_threshold" default="-2.1" />-->
    <arg name="distance_threshold" default="0.20" />
    <arg name="map_tf" default="velo_middle" />
    <arg name="lowest_segment_point_num" default="6" />


    <node pkg="ground_detect" type="lidar_line_analyse" name="lidar_line_analyse_node" output="screen">
        <remap from="/lidar/vlp32_middle/PointCloud2" to="/lidar/vlp32_middle/PointCloud2" />
        <param name="low_lines" value="$(arg low_lines)" />
<!--        <param name="z_threshold" value="$(arg z_threshold)" />-->
<!--        <param name="distance_threshold" value="$(arg distance_threshold)" />-->
    </node>

    <node pkg="ground_detect" type="feature_based_select" name="feature_based_select_node" output="screen">
        <param name="low_lines" value="$(arg low_lines)" />
        <param name="z_threshold" value="$(arg z_threshold)" />
        <param name="distance_threshold" value="$(arg distance_threshold)" />
        <param name="lowest_segment_point_num" value="$(arg lowest_segment_point_num)" />
    </node>

    <node pkg="ground_detect" type="path_provider" name="path_provider_node" output="screen">
        <param name="low_lines" value="$(arg low_lines)" />
    </node>
    <!--rviz-->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find ground_detect)/rviz/result.rviz -f $(arg map_tf)" />
</launch>
